manager/apis/app/v1alpha1/fybrikmodule_types.go

```diff
@@ -223,6 +224,21 @@ type ChartSpec struct {
 	Values map[string]string `json:"values,omitempty"`
 }
 
+// FybrikModuleStatus defines the observed state of FybrikModule.
+type FybrikModuleStatus struct {
+	// ErrorMessage indicates that an error has happened during the module validation
+	// +optional
+	ErrorMessage string `json:"errorMessage,omitempty"`
+
+	// ValidatedGeneration is the version of the FybrikModule that has been validated with the taxonomy defined.
+	// +optional
+	ValidatedGeneration int64 `json:"validatedGeneration,omitempty"`
+
+	// ValidModule indicates whether the FybrikModule is valid given the defined taxonomy
+	// +optional
+	ValidModule corev1.ConditionStatus `json:"validModule,omitempty"`
```

Comment:
@shlomitk1 is that aligned with how we usually handle readiness and errors? Should `errorMessage` be just `error`? Should `validModule` be `ready`? Should conditions be used instead? (the use of `corev1.ConditionStatus` is not within a condition).

Reply:
We could reuse the following structure that is currently defined in fybrikapplication_types.go
```
type Condition struct {
	// Type of the condition
	Type ConditionType `json:"type"`
	// Status of the condition: true or false
	Status corev1.ConditionStatus `json:"status"`
	// Message contains the details of the current condition
	// +optional
	Message string `json:"message,omitempty"`
}
```
We could even leverage the same types of Deny, Ready and Error (Deny == invalid, Ready == valid, Error == could not validate).

Reply:
@roee88 `ErrorMessage` and `ValidModule` are consistent with the `Status` fields we defined in [FybrikApplication](https://github.com/fybrik/fybrik/blob/895cc922d74d10c02f229caa291e0d41918d13f7/manager/apis/app/v1alpha1/fybrikapplication_types.go#L182)